### PR TITLE
bpf: Fix default configuration of bpf/

### PR DESCRIPTION
**Pull request by joestringer**
The default configuration of binaries compiled in the bpf/ directory was
broken in commit 84f82e7ccd05 ("agent: Make IPv6 optional"). This causes
the complexity checker scripts to severely underestimate the datapath
complexity, breaking the verifier complexity tests. (Breaking in that they're
ineffective, so they will never complain).

Rearrange the logic so that the default configuration will enable all
possible code, then specific parts may be conditionally disabled.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6484)

### CODE REVIEW

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
#6082 was marked for backport to 1.4, so I've marked this also.

File: bpf/Makefile [hunk 1]
```diff
@@ -46,10 +46,10 @@ LXC_OPTIONS = \
...
 	 -DHAVE_LPM_MAP_TYPE \
 	 -DHAVE_LRU_MAP_TYPE \
-	 -DENABLE_IPV4 \
-	 -DENABLE_IPV6 \
-	 "-DENABLE_IPV6 -DENABLE_IPV4" \
-	 "-DENABLE_IPV4 -DENABLE_NAT46"
+	 -DDISABLE_IPV4 \
+	 -DDISABLE_IPV6 \
+	 "-DDISABLE_IPV6 -DDISABLE_IPV4" \
+	 "-DDISABLE_IPV4 -DENABLE_NAT46"
```
Comment (joestringer) [inline bpf/Makefile]: NAT46 also needs this treatment, it used to be unconditionally enabled and now it's not.

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
By the way, the lines in this makefile which attempt to combine multiple options do not work as intended (#2906).

**Review by tgraf — APPROVED** [review]

**Pull request merged**